lib/reader/src/error.rs

```diff
@@ -19,6 +19,8 @@ pub struct Error {
     pub location: Location,
     /// Error message.
     pub message: String,
+    /// Error because of unsupported ISA
+    pub unsupported: bool,
```

Comment:
Hmm. I think it shouldn't be an error if a test file contains an unsupported ISA.

A test file [can contain multiple ISA specs](http://cretonne.readthedocs.io/en/latest/testing.html#file-tests) and it is possible that some of them are supported while others are unsupported. In that case, we should run the test on the supported ISAs and ignore the unsupported ISAs.

I think the `IsaSpec` enum can represent everything we need:

- The file has no ISA specs: Use `IsaSpec::None(..)`.
- The file has ISA specs, but none are supported: Use `IsaSpec::Some([])`.
- The file has ISA specs, not all supported: Use `IsaSpec::Some([a, b, ...])` with just the supported ISAs. Any unsupported ISA specs are ignored and omitted.

It should still be a parser error if we see an *unknown* ISA.